### PR TITLE
ARROW-10002: [Rust] Remove `default fn` from `PrimitiveArrayOps`

### PR DESCRIPTION
**Pull request by BatmanAoD**
[Problem description](https://github.com/BatmanAoD/arrow-rust-specialization-alternatives/blob/master/README.md#problem-summary-using-primitivearrayops)

This is based on my [proof-of-concept](https://github.com/BatmanAoD/arrow-rust-specialization-alternatives/pull/1) solution.

### CODE REVIEW

File: rust/arrow/src/datatypes.rs [hunk 1]
```diff
@@ -208,12 +210,22 @@ pub trait ArrowPrimitiveType: 'static {
     fn get_data_type() -> DataType;
 
     /// Returns the bit width of this primitive type.
-    fn get_bit_width() -> usize;
+    fn get_bit_width() -> usize {
+        size_of::<Self::Native>()
```
Comment (BatmanAoD) [inline rust/arrow/src/datatypes.rs:214]:
```suggestion
        size_of::<Self::Native>() * 8
```

**Review by andygrove — APPROVED** [review]
Thanks @BatmanAoD this LGTM but I would like to see reviews from other other Rust contributors before merging this, in case there are subtle details that I am missing. Perhaps @paddyhoran or @nevi-me could take a look too?

**Review by paddyhoran — APPROVED** [review]
Nice. LGTM.  Thanks @BatmanAoD

**Comment by andygrove** [comment]
@BatmanAoD Could you rebase against master (not merge) and I can merge this.

**Comment by andygrove** [comment]
Test failures are unrelated (conda integration tests)

**Comment by jorgecarleitao** [comment]
🎉  this is highly appreciated, and the API is also easier to work with.

**Comment by sunchao** [comment]
Just found out this... Nice work @BatmanAoD !

**Pull request closed without merging**